src/control/conf.c

```diff
@@ -259,10 +264,70 @@ gchar *dt_conf_get_string(const char *name)
   return g_strdup(str);
 }
 
+static char *_sanitize_confgen(const char *name, const char *value)
+{
+  const dt_confgen_value_t *item = g_hash_table_lookup(darktable.conf->x_confgen, name);
+
+  if(!item) return g_strdup(value);
+
+  char result[100] = { 0 };
+  g_strlcpy(result, value, sizeof(result));
+
+  switch(item->type)
+  {
+    case DT_INT:
+    {
+      float v = dt_calculator_solve(1, value);
+
+      const int min = item->min ? (int)dt_calculator_solve(1, item->min) : INT_MIN;
+      const int max = item->max ? (int)dt_calculator_solve(1, item->max) : INT_MAX;
+      // if garbadge, use default
+      const int val = isnan(v) ? dt_confgen_get_int(name, DT_DEFAULT) : (int)v;
+      l_sprintf(result, sizeof(result), "%d", CLAMP(val, min, max));
```

Comment:
I can't compile dt with this change, here's the error:
```
/home/johnny/darktable_repos/darktable/src/control/conf.c: In function ‘_sanitize_confgen’:
/home/johnny/darktable_repos/darktable/src/control/conf.c:286:7: error: implicit declaration of function ‘l_sprintf’; did you mean ‘g_sprintf’? [-Werror=implicit-function-declaration]
  286 |       l_sprintf(result, sizeof(result), "%d", CLAMP(val, min, max));
      |       ^~~~~~~~~
      |       g_sprintf
```

Reply:
Should be fixed, a copy&paste error, this is intended to be used only for lua code and I suppose you don't have lua activated on your side.